code/modules/mob/living/simple_animal/abnormality/he/white_lake.dm

```diff
@@ -87,15 +88,34 @@
 /obj/item/ego_weapon/flower_waltz
 	name = "waltz of the flowers"
 	desc = "It's awfully fun to write a march for tin soldiers, a waltz of the flowers."
+	special = "Cannot be dropped until moved from your hands."
 	icon_state = "flower_waltz"
 	force = 18
 	damtype = PALE_DAMAGE
 	armortype = PALE_DAMAGE
 	attack_verb_continuous = list("slices", "cuts")
 	attack_verb_simple = list("slices", "cuts")
 	hitsound = 'sound/weapons/bladeslice.ogg'
+	var/unequipped = FALSE
 	//No requirements because who knows who will use it.
 
+// Sets the weapon to not be droppable until it's moved from the main hand. No more leg sweeping.
+/obj/item/ego_weapon/flower_waltz/equipped(mob/user, slot, initial = FALSE)
+	.=..()
+	if (unequipped)
+		return
+	if (slot == ITEM_SLOT_HANDS && !unequipped)
+		RegisterSignal(user, COMSIG_LIVING_DEATH, .proc/Champion_Death)
+		ADD_TRAIT(src, TRAIT_NODROP, src)
```

Comment:
This all should probably be done in WL's code when it's put into champion's hands.
It will prevent admin-spawned sword from being undroppable.

Reply:
Due to the variable starting as False, it should never be truly undroppable, as it would simply take a swap to belt/suit storage to permanently remove the trait from it. I felt having all the equip and undroppable code in one spot as well was the best choice, otherwise it would've been entirely the removal code here and entirely the adding code in the champion application.

Reply:
NODROP trait prevents it from being moved anywhere, including backpack, pockets and any other slots on you.

Reply:
it does not prevent it from being belt-slotted or suit-slotted. I've tested this.